File: server/dotnet/sdk-dotnet-example/Business/Handlers/TokenPaymentHandler.cs
using Business.Domain.Enums.Payments;
using Business.DTOs.CreatePayments;
using Business.Interfaces.Handlers;
using Business.Interfaces.SDKClients;

namespace Business.Handlers;

public class TokenPaymentHandler(IPaymentClient paymentClient) : IPaymentMethodHandler
{
    public PaymentMethodType SupportedMethod => PaymentMethodType.TOKEN;

    public async Task<CreatePaymentResponseDto> HandleAsync(CreatePaymentRequestDto request)
    {
        return await paymentClient.CreatePaymentAsync(request);
    }
}

File: server/ruby/app/business/handlers/token_payment_handler.rb
module Business
  module Handlers
    class TokenPaymentHandler
      def initialize(payment_client:)
        @payment_client = payment_client
      end

      def supported_method
        Domain::Enums::Payments::PaymentMethodType::TOKEN
      end

      def handle(request_dto)
        @payment_client.create_payment(request_dto)
      end
    end
  end
end
